gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -1370,6 +1377,58 @@ private static long getMaxRemainingResults(
     return maxResults - numResults;
   }
 
+  /**
+   * TODO(hgong): Maybe a pre-filter can be done to filter out only the keys that match certain
+   * object name prefix pattern.
+   */
+  private String getMatchedObjectNamePrefix(Map<String, String> prefixes, String targetPrefix) {
```

Comment:
Prefix-based auth seems a very different from user/group-auth in terms of implementation. May you implement it in separate follow up PR? It will be easier to reason about implementation this way.

Reply:
reverted the change, but kept the prefix configuration property name to make follow up change easier.